lib/receptor_controller/client/response_worker.rb

```diff
@@ -105,14 +105,21 @@ def process_message(message)
             # Response OK
             #
             message_type = response['message_type'] # "response" (with data) or "eof" (without data)
-            registered_messages.delete(message_id) if message_type == 'eof'
-
             payload = response['payload']
-            payload = unpack_payload(payload) if message_type == 'response' && payload.kind_of?(String)
 
-            logger.debug("Receptor response: OK | message #{message_id} (#{payload})")
+            if message_type == 'eof'
+              registered_messages.delete(message_id)
```

Comment:
@lindgrenj6 Wont this delete the item from the hash as soon as we get the eof. This can only happen after we have recd. all messages

Reply:
@mkanoor updated it to do the checks other than this.